File: youyougou-manager-web/src/main/webapp/js/controller/brandController.js
/**
 * 品牌前端控制层
 */
app.controller('brandController',function($scope,$http,$controller,brandService) {

    $controller('baseController',{$scope:$scope});

    //查询品牌列表
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;//给列表变量赋值
            }
        );
    }

    //分页
    $scope.findPage=function(page,size){
        brandService.findPage(page,size).success(
            function(response){
                $scope.list=response.rows;//显示当前页数据
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

    //增加
    $scope.save=function () {
        var object=null;

        if($scope.entity.id!=null){
            object=brandService.update($scope.entity);//执行修改方法
        }else {
            object=brandService.add($scope.entity);
        }

        object.success(
            function (response) {
                if(response.success){
                    //重新加载
                    $scope.reloadList();
                }else {
                    alert(response.message);
                }
            }
        );
    }
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        );
    }


    //批量删除
    $scope.delete=function () {
        //获取选中的复选框
        brandService.delete($scope.selectIds).success(
            function(response){
                if(response.success){
                    $scope.reloadList();//刷新列表
                }
            }
        );
    }

    //定义查询实体
    $scope.searchEntity={};
    //条件查询
    $scope.search=function (page,size) {
        brandService.search(page,size,$scope.searchEntity).success(
            function (response) {
                $scope.paginationConf.totalItems=response.total;//总记录数
                $scope.list=response.rows;//给列表变量赋值
            }
        );
    }
});

File: youyougou-manager-web/src/main/java/com/youyougou/manager/controller/BrandController.java
package com.youyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.youyougou.entity.PageResult;
import com.youyougou.entity.Result;
import com.youyougou.pojo.TbBrand;
import com.youyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * 品牌管理业务逻辑层
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 查询所有品牌数据
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    /**
     * 分页查询所有品牌数据
     * @param page
     * @param size
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int size){
        System.out.println("hello");
        return brandService.findPage(page,size);
    }

    /**
     * 添加品牌
     * @param tbBrand
     * @return 返回添加结果
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){
        try {
            brandService.add(tbBrand);
            return new Result(true,"增加成功");

        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"增加失败");
        }
    }

    /**
     * 更新数据
     * @param brand
     * @return 返回更新结果
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try{
            brandService.update(brand);
            return new Result(true,"修改成功");

        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /**
     * 查找实体对象
     * @param id
     * @return 返回单个品牌
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        if(ids==null||ids.length==0){
            return new Result(false,"请选择你要删除的记录");
        }
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    /**
     * 查询加分页
     * @param brand
     * @param page
     * @param size
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,int page,int size){
       return   brandService.findPage(brand,page,size);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }


}
